### PR TITLE
fiddler: Make the script editor work

### LINKED ISSUE #4225: fiddler: The Fiddler ScriptEditor won't work
With the change of #2322, the fiddler's script editor won't work as well as the *Customize Rules* function.

### PR DESCRIPTION
**Pull request by sinloss**
- Closes #4225

The registry keys are based on https://feedback.telerik.com/fiddler/1361507-failed-binding-error-when-selecting-customize-rules

### CODE REVIEW

File: bucket/fiddler.json [hunk 1]
```diff
@@ -6,15 +6,25 @@
...
+        "script": "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\FiddlerSetup.exe\" \"$dir\" -Removal",
+        "Move-Item \"$dir\\FSE2.exe\" \"$dir\\ScriptEditor\\\"",
+			"Set-Location -Path 'HKCU:\\Software\\Microsoft'",
+			"New-Item -Name 'Fiddler2\\InstallerSettings' -Force",
+			"Set-Location -Path 'Fiddler2\\InstallerSettings'",
+			"New-ItemProperty -Path '.' -Name 'InstallPath' -Value \"$dir\\\" -PropertyType ExpandString -Force",
+			"New-ItemProperty -Path '.' -Name 'PluginPath' -Value \"$dir\\Inspectors\\\" -PropertyType ExpandString -Force",
+			"New-ItemProperty -Path '.' -Name 'ScriptPath' -Value \"$dir\\Scripts\\\" -PropertyType ExpandString -Force"
     },
     "post_install": "Remove-Item \"$dir\\`$PLUGINSDIR\" -Recurse",
```
Comment (Ash258) [inline bucket/fiddler.json:18]:
```suggestion
    "pre_install": [
        "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\FiddlerSetup.exe\" \"$dir\" -Removal",
        "Remove-Item \"$dir\\`$*\" -Recurse",
        "    # Script editor",
        "Move-Item \"$dir\\FSE2.exe\" \"$dir\\ScriptEditor\"",
        "$path = 'HKCU:\\Software\\Microsoft\\Fiddler2\\InstallerSettings'",
        "New-Item $path -ErrorAction SilentlyContinue -Force | Out-Null",
        "@(@('InstallPath', ''), @('PluginPath', 'Inspectors'), @('ScriptPath', 'Scripts')) | ForEach-Object {",
        "    New-ItemProperty -Path $path -Name $_[0] -Value \"$dir\\$($_[1])\" -PropertyType ExpandString -ErrorAction SilentlyContinue | Out-Null",
        "}"
    ],
    "uninstaller": {
        "script": "if ($purge) { Remove-Item 'HKCU:\\Software\\Microsoft\\Fiddler2' -Force -Recurse }"
    },
```

File: bucket/fiddler.json [hunk 2]
```diff
@@ -6,15 +6,25 @@
...
     "shortcuts": [
         [
             "Fiddler.exe",
             "Fiddler"
-        ]
+        ], [
+			"ScriptEditor\\FSE2.exe",
+			"Fiddler ScriptEditor"
+		]
     ],
```
Comment (Ash258) [inline bucket/fiddler.json:28]:
```suggestion
        ],
        [
            "ScriptEditor\\FSE2.exe",
            "Fiddler ScriptEditor"
        ]
    ],
    "persist": "Scripts",
```

**Comment by sinloss** [comment]
> > @niheaven were you reviewing this?
> 
> No, I just resolve conflicts.
> 
> @sinloss
> 
> image

Sorry, but I cannot reproduce this on my machine which is -
```
Edition: Windows 10 Pro 19042.1288
Version: 20H2
OS build: 19042.1288
```

Is there any chance that you may provide some details?

File: bucket/fiddler.json [hunk 3]
```diff
@@ -5,17 +5,31 @@
...
+        "New-Item $path -ErrorAction SilentlyContinue -Force | Out-Null",
+        "@(@('InstallPath', ''), @('PluginPath', 'Inspectors'), @('ScriptPath', 'Scripts')) | ForEach-Object {",
+        "    New-ItemProperty -Path $path -Name $_[0] -Value \"$dir\\$($_[1])\" -PropertyType ExpandString -ErrorAction SilentlyContinue | Out-Null",
+        "}"
+    ],
+    "uninstaller": {
+        "script": "If ($purge) { Remove-Item 'HKCU:\\Software\\Microsoft\\Fiddler2' -Force -Recurse }"
     },
-    "post_install": "Remove-Item \"$dir\\`$PLUGINSDIR\" -Recurse",
-    "bin": "Fiddler.exe",
```
Comment (pratikpc) [inline bucket/fiddler.json:12]: Is there a reason for removing bin?

Comment (sinloss) [reply bucket/fiddler.json:12]: It's been my mistake, thanks for pointing it out. I'll bring the `bin` back...

**Comment by niheaven** [comment]
@pratikpc @rashil2000 Could you please test this app? Maybe the bug is only occurred on my machine.

**Comment by rashil2000** [comment]
Works on Win 10

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**